### PR TITLE
Provide suggestion for replacing deprecated RUNPATH %d specifiers

### LINKED ISSUE #4543: Add feedback in suggester for deprecated runpath syntax
Relates to #4432

We should inform the user whenever deprecated syntax for runpath in found in the configuration files.
E.g. 
``` python
realization-%d/iteration-%d # deprecated
realization-<IENS>/iteration-<ITER>
```

### PR DESCRIPTION
**Pull request by andreas-el**
**Issue**
Resolves #4543 

Unsure if this should be 'improvement' or 'maintenance'

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Review by oysteoh — APPROVED** [review]

File: src/ert/_c_wrappers/enkf/_deprecation_migration_suggester.py [hunk 1]
```diff
@@ -47,6 +47,13 @@ def add_suggestion(kw, suggestion):
                 logger.info("Deprecated keyword %s", kw)
                 suggestions.append(suggestion)
 
+        if content.hasKey("RUNPATH") and "%d" in content.getValue("RUNPATH"):
+            add_suggestion(
+                "RUNPATH",
+                "The use of %d in RUNPATH has been deprecated. "
+                "Instead use equivalent RUNPATH syntax: realization-<IENS>/iter-<ITER>",
```
Comment (eivindjahren) [inline src/ert/_c_wrappers/enkf/_deprecation_migration_suggester.py:54]: Is "syntax" too technical here? Should we try something simpler like "Instead use the <IENS> <ITER> keywords, e.g.:  realization-<IENS>/iter-<ITER>"

**Review by eivindjahren — CHANGES_REQUESTED** [review]

File: src/ert/_c_wrappers/enkf/_deprecation_migration_suggester.py (hunk 1 above)
Comment (andreas-el) [reply src/ert/_c_wrappers/enkf/_deprecation_migration_suggester.py:54]: Np. :+1:

**Review by eivindjahren — CHANGES_REQUESTED** [review]
Sorry for not catching this issue in the first iteration of reviews: could you also add a test for this in  `test_config_parsing/test_suggester.py`.

**Review by eivindjahren — APPROVED** [review]
:brain:

**Pull request merged**